src/components/Button.js

```diff
@@ -76,6 +79,7 @@ const defaultProps = {
     isDisabled: false,
     small: false,
```

Comment:
This is NAB because it's unlikely to be an actual problem, but I think in general it's a bad practice to have mutually exclusive props – it just leads to less-predictable code. For example, what is the expected behavior if I pass both `large` and `medium` props to the `Button` component?

So in this case I think a better pattern would be to have a single `size` prop that's `PropTypes.oneOf(['small', 'medium', 'large'])` (and create a `CONST.BUTTON_SIZE` constant to go with it).

Thoughts?

Reply:
> in general it's a bad practice to have mutually exclusive props – it just leads to less-predictable code

Well, I agree about the latter but not the former. While size is a good suggestion, I feel we should fix how to deal with mutually exclusive props. For me as a developer, it is easier to write `large`, than import `CONST` and then specifying `size={CONST.BUTTON_SIZE.LARGE}`.

But I can change it to whatever you'd prefer.

Reply:
Yeah, Boolean props are the way to go for most of the general-purpose React components. I don't prefer `PropTypes.oneOf(['small', 'medium', 'large'])`. 
My rule:
1. If we are defining variants of the component then use boolean props `small or isDisabled` etc. 
2. If we are changing the behavior of the component then use value-props `prop={value}`;

Reply:
So I'll start by saying this is still NAB for me, because nobody is likely to pass both `large` and `small` to the same component. So if either of you still feel strongly, feel free to just resolve this thread.

> For me as a developer, it is easier to write large, than import CONST and then specifying size={CONST.BUTTON_SIZE.LARGE}.

I agree that `<Button large />` is prettier than `<Button size={CONST.BUTTON_SIZE.LARGE} />`, but I don't think that makes it a better pattern. Isn't a consistent and predictable interface for the component more important than saving developer keystrokes by not having to import a constant?

> Yeah, Boolean props are the way to go for most of the general-purpose React components...If we are defining variants of the component then use boolean props small or isDisabled etc....If we are changing the behavior of the component then use value-props prop={value};

This seems like a pretty subtle distinction to me, and I'm not sure I understand the motivations behind it. In my opinion, props should be defined based on the data they represent.

> If we are defining variants of the component then use boolean props small or isDisabled etc.

Following along with this example, I see an important difference between these two props. `isDisabled` is truly boolean/binary in that no other prop conflicts with it – a button is either `isDisabled` or it is not. Meanwhile, `small` directly conflicts with `large` – passing both makes no sense. What we really have here is not binary (boolean) data, but trinary data (a button is small, medium, or large).

In your slider example, I would create a single `horizontal` prop – then we have boolean data. The other three props (speed, animation, number of items) don't represent boolean data. `Speed` is most likely a floating point number (continuous data), `animation` is most likely one of several distinct strings (X-nary data), and `number of items` is a positive integer.

Reply:
> Following along with this example, I see an important difference between these two props. isDisabled is truly boolean/binary in that no other prop conflicts with it – a button is either isDisabled or it is not. Meanwhile, small directly conflicts with large – passing both makes no sense. What we really have here is not binary (boolean) data, but trinary data (a button is small, medium, or large).

This suddenly started to sound meaningful :rofl:. I change my mind. But let's skip this change for this PR. We can discuss this as a style pattern to follow on separately. Maybe create a Planning issue as Marc use to do and continue there.

Reply:
I am guessing changes will require updating the Button props across the application. The same would have to be done for `success`, etc. too if needed.

Please let me know the final thing, accordingly, we can then close this, or I can make changes.

Reply:
Okay, not sure I'll ever get around to it but I created a separate issue https://github.com/Expensify/App/issues/7220